metrics/handler.go

```diff
@@ -0,0 +1,79 @@
+package metrics
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/datatrails/go-datatrails-common/tenantid"
+)
+
+// Tailored Prometheus metrics
+const (
+	URLPrefix = "archivist"
+)
+
+// we have to intercept the ResponseWriter in order to get the statuscode
+type LoggingResponseWriter struct {
+	http.ResponseWriter
+	StatusCode int
+}
+
+func (m *Metrics) NewLatencyMetricsHandler(h http.Handler) http.Handler {
+
+	if m == nil {
+		return h
+	}
+	m.log.Debugf("NewLatencyMetricsHandler")
+	observer := NewLatencyObservers(m)
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		log := m.log.FromContext(r.Context())
+		defer log.Close()
+
+		fields := strings.Split(strings.Trim(r.URL.Path, "/ "), "/")
```

Comment:
does the url package give us this without having to do strings.Split and strings.Trim?

Reply:
I dont think so - once you have the Path thendoing a split seems to be the accepted method....